haystack/pipeline.py

```diff
@@ -1287,4 +1288,31 @@ def run(self, query, documents, **kwargs):
         # Pass also the other incoming kwargs so that future nodes still have access to it
         output.update(**kwargs)
 
-        return output, "output_1"
\ No newline at end of file
+        return output, "output_1"
+
+
+class MostSimilarDocumentsPipeline(BaseStandardPipeline):
+    def __init__(self, document_store: BaseDocumentStore):
+        """
+        Initialize a Pipeline for finding similar documents using document ids.
+
+        :param document_store: Document Store instance
+        """
+        self.document_store = document_store
+
+    def run(self, document_ids: List[str], top_k: int = 5):
+        """
+        :param document_ids: document ids
+        :param top_k: How many documents id to return against single document
+        """
+        return_ids: list = []
+        self.document_store.return_embedding = True  # type: ignore
+        for document in self.document_store.get_documents_by_id(ids=document_ids):
+            found_ids: list = []
+            for found_document in self.document_store.query_by_embedding(query_emb=document.embedding,
+                                                                         return_embedding=False,
+                                                                         top_k=top_k):
+                found_ids.append(found_document.id)
+            return_ids.append(found_ids)
```

Comment:
Let's rather return a list of Documents here than just the ids

Reply:
Done.